sklearn/metrics/scorer.py

```diff
@@ -279,6 +289,9 @@ def make_scorer(score_func, greater_is_better=True, needs_proba=False,
         For example ``average_precision`` or the area under the roc curve
         can not be computed using discrete predictions alone.
 
+    name : str, default=None
+        The name for the scorer. If None, the score_func.__name__ is used.
```

Comment:
I'm ambivalent. I'm fine with keeping it easy for a user to make a scorer. If we made a public `register_scorer` then that might change things. But I think that's a separate PR.